### PR TITLE
Unite most of tarpc into a single crate

### PR DESCRIPTION
**Pull request by vorot93**
This follows in the footsteps of Tokio. It should make maintenance a lot easier while retaining the flexibility that users may want.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,13 +41,13 @@ Some other features of tarpc:
   sends a request to another server, that server will see an 8s deadline.
 - Serde serialization: enabling the `serde1` Cargo feature will make service requests and
   responses `Serialize + Deserialize`. It's entirely optional, though: in-memory transports can
-  be used, as well, so the price of eerialization doesn't have to be paid when it's not needed.
+  be used, as well, so the price of serialization doesn't have to be paid when it's not needed.
```
Comment (tikue) [inline README.md]: I thought I fixed this, but I only changed the crate documentation :)

File: tarpc/Cargo.toml [hunk 2]
```diff
@@ -13,28 +13,49 @@ readme = "../README.md"
...
+pin-project = "0.4"
+raii-counter = "0.2"
+rand = "0.7"
+tokio-timer = "0.3.0-alpha"
+serde = { optional = true, version = "1.0", features = ["derive"] }
+tokio = { optional = true, version = "0.2.0-alpha" }
+tarpc-plugins = { path = "../plugins" }
+
+async-bincode = { optional = true, version = "0.4" }
+futures-legacy = { optional = true, version = "0.1", package = "futures" }
```
Comment (tikue) [inline tarpc/Cargo.toml]: What still uses this?

File: tarpc/src/rpc/client/channel.rs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 // license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
+use crate::trace::SpanId;
```
Comment (tikue) [inline tarpc/src/rpc/client/channel.rs]: Let's put this in the `use` block on the next line down.

File: README.md [hunk 4]
```diff
@@ -41,13 +41,13 @@ Some other features of tarpc:
...
 
 ### Usage
 Add to your `Cargo.toml` dependencies:
 
 ```toml
-tarpc = "0.18.0"
+tarpc = { version = "0.18.0", features = ["full"] }
```
[the remark this answers is not in the document]
Comment (vorot93) [reply README.md]: We should but only after all other PRs are in

File: tarpc/Cargo.toml (hunk 2 above)
Comment (vorot93) [reply tarpc/Cargo.toml]: bitcode transport

**Pull request merged**